docassemble/assemblylinewizard/interview_generator.py

```diff
@@ -887,97 +891,140 @@ def get_docx_variables( text ):
+              reserved_var_plurals=generator_constants.RESERVED_VAR_PLURALS,
+              reserved_pluralizers_map = generator_constants.RESERVED_PLURALIZERS_MAP,
+              reserved_suffixes_map=generator_constants.RESERVED_SUFFIXES_MAP):
+    """For a given set of specific cases, transform a
+    PDF field name into a standardized object name
+    that will be the value for the attachment field."""
+    if document_type.lower() == "docx":
+        return label # don't transform DOCX variables
 
-  if exactly_matches_reserved_word(reserved_whole_words, label):
-    return label
+    # Remove multiple appearance indicator, e.g. '__4' of 'users__4'
+    label = remove_multiple_appearance_indicator(label)
 
-  # For the sake of time, this is the fastest way to get around something being plural (wrap it in `str()`)
-  if is_a_plural(reserved_var_plurals, label):
-    return get_stringifiable_version(label)
+    if label in reserved_whole_words:
+        return label
+
+    if label in reserved_var_plurals:
+        return label
+
+    # Break up label into its parts
+    label_groups = get_reserved_label_parts(reserved_prefixes, label)
+
+    # If no matches to automateable labels were found,
+    # just use the label as it is
+    if label_groups is None or label_groups[1] == '':
+        return label
+
+    # With reserved words, we're always using an index
+    # of the plural version of the prefix of the label
+    prefix = label_groups[1]
+    # Turn any singluars into plurals, e.g. 'user' into 'users'
+    plural_prefix = reserved_pluralizers_map[prefix]
+    digit = label_groups[2]
+    index = indexify(digit)
+    # it's just a standalone, like "defendant", or it's a numbered singular
+    # prefix, e.g. user3
+    if label == prefix or label == prefix + digit: 
+        return plural_prefix + index # the vars w/ the index, users[0] or users[2]
+
+    suffix = label_groups[3]
+    # Avoid transforming arbitrary suffixes into attributes
+    if not suffix in reserved_suffixes_map:
+        return label  # return it as is
+
+    # Get the mapped suffix attribute if present, else just use the same suffix
+    suffix_as_attribute = reserved_suffixes_map.get(suffix, suffix)
+    return "".join([plural_prefix, index, suffix_as_attribute])
+
+
+def trigger_gather_string(docassemble_var, 
+                          docx_only_suffixes=generator_constants.DOCX_ONLY_SUFFIXES,
+                          reserved_whole_words=generator_constants.RESERVED_WHOLE_WORDS,
+                          reserved_var_plurals=generator_constants.RESERVED_VAR_PLURALS,
+                          reserved_pluralizers_map=generator_constants.RESERVED_PLURALIZERS_MAP,
+                          reserved_suffixes_map=generator_constants.RESERVED_SUFFIXES_MAP):
+    """For a given set of specific cases, transform a
+    PDF field name into a standardized object name
+    that will be the value for the attachment field."""
```

Comment:
To clarify, `.gather()` is not for the attachment field. That's for the interview order code block. This might just be an out of date function description, but I wanted to make sure we were on the same page.

Reply:
We are on the same page, that is just copy-pasted code that I never updated to the new function.  Thanks for finding!